Update gluster provisioner parameters

wongma7 opened the pull request:
make it match https://github.com/kubernetes/kubernetes/blob/release-1.4/examples/experimental/persistent-volume-provisioning/README.md

@jsafrane

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/kubernetes/kubernetes.github.io/1226)
<!-- Reviewable:end -->

devin-donnelly commented:
Is this not relevant until 1.4 is released? I'm noticing you're referencing something in release-1.4.

wongma7 commented:
@devin-donnelly yes, so this should be targeting the release-1.4 branch right? It looks like a lot of stuff I did previously that was intended only for 1.4 is in master, so I will fix that in a separate PR...

devin-donnelly commented:
Yes, correct. This should target release-1.4.

devin-donnelly commented on docs/user-guide/persistent-volumes/index.md:
```diff
@@ -365,22 +365,24 @@ parameters:
-* `resturl` : Gluster REST service url which provision gluster volumes on demand. The format should be a valid URL and this is a mandatory parameter for GlusterFS dynamic provisioner.
-* `restuser` : Gluster REST service user who has access to create volumes in the Gluster Trusted Pool. This parameter is optional, empty string will be used when omitted.
-* `restuserkey` : Gluster REST service user's password which will be used for authentication to the REST server. This parameter is optional, empty string will be used when omitted.
+* `resturl` : Gluster REST service url which provision gluster volumes on demand. The format should be `IPaddress:Port` and this is a mandatory parameter for GlusterFS dynamic provisioner.
```
provision -> provisions

"this parameter is mandatory when using the GlusterFS dynamic provisioner."

devin-donnelly commented on docs/user-guide/persistent-volumes/index.md:
```diff
@@ -365,22 +365,24 @@ parameters:
-* `restuser` : Gluster REST service user who has access to create volumes in the Gluster Trusted Pool. This parameter is optional, empty string will be used when omitted.
-* `restuserkey` : Gluster REST service user's password which will be used for authentication to the REST server. This parameter is optional, empty string will be used when omitted.
+* `resturl` : Gluster REST service url which provision gluster volumes on demand. The format should be `IPaddress:Port` and this is a mandatory parameter for GlusterFS dynamic provisioner.
+* `restauthenabled` : Gluster REST service authentication boolean is required if the authentication is enabled on the REST server. If this value is 'true', 'restuser' and 'restuserkey' have to be filled.
```
Needs a clarity edit:

"A boolean value that indicates whether Gluster REST service authentication is enabled on the REST server. If this value is 'true', you must supply values for the 'restuser' and 'restuserkey' parameters."

devin-donnelly commented on docs/user-guide/persistent-volumes/index.md:
```diff
@@ -365,22 +365,24 @@ parameters:
-* `restuserkey` : Gluster REST service user's password which will be used for authentication to the REST server. This parameter is optional, empty string will be used when omitted.
+* `resturl` : Gluster REST service url which provision gluster volumes on demand. The format should be `IPaddress:Port` and this is a mandatory parameter for GlusterFS dynamic provisioner.
+* `restauthenabled` : Gluster REST service authentication boolean is required if the authentication is enabled on the REST server. If this value is 'true', 'restuser' and 'restuserkey' have to be filled.
+* `restuser` : Gluster REST service user who has access to create volumes in the Gluster Trusted Pool.
```
Comma after "user".

devin-donnelly commented on docs/user-guide/persistent-volumes/index.md:
```diff
@@ -365,22 +365,24 @@ parameters:
+* `resturl` : Gluster REST service url which provision gluster volumes on demand. The format should be `IPaddress:Port` and this is a mandatory parameter for GlusterFS dynamic provisioner.
+* `restauthenabled` : Gluster REST service authentication boolean is required if the authentication is enabled on the REST server. If this value is 'true', 'restuser' and 'restuserkey' have to be filled.
+* `restuser` : Gluster REST service user who has access to create volumes in the Gluster Trusted Pool.
+* `restuserkey` : Gluster REST service user's password which will be used for authentication to the REST server.
```
delete "which", comma after "password".

wongma7 commented:
I changed target branch and addressed the feedback in the same commit, looks like it broke github review though. Let me know if I should just open a new PR..

devin-donnelly commented:
No need, that looks fine. Docs LGTM.

devin-donnelly commented:
@mikedanese , can you or someone else give this a tech review? It should be short.

jsafrane commented:
I can review it. There is one minor issue with address format, I'd rather have it precise.

devin-donnelly commented:
@wongma7 , can you ping me when you've addressed @jsafrane 's feedback? Thanks.

wongma7 commented:
@devin-donnelly addressed feedback, made format more precise

jsafrane commented:
lgtm from tech side
